source/core/slang-process.h

```diff
@@ -0,0 +1,73 @@
+// slang-process.h
+#ifndef SLANG_PROCESS_H
+#define SLANG_PROCESS_H
+
+#include "slang-string.h"
+#include "slang-list.h"
+#include "slang-stream.h"
+#include "slang-io.h"
+
+#include "slang-string-escape-util.h"
+
+#include "slang-command-line.h"
+
+namespace Slang {
+
+class Process : public RefObject
+{
+public:
+    enum class StreamType
+    {
+        ErrorOut,
+        StdOut,
+        StdIn,
+        CountOf,
+    };
+
+    typedef uint32_t Flags;
+    struct Flag
+    {
+        enum Enum : Flags
+        {
+            AttachDebugger = 0x01,
+        };
+    };
+
+        /// Get the stream for the type
+    Stream* getStream(StreamType type) const { return m_streams[Index(type)]; }
+    int32_t getReturnValue() const { return m_returnValue;  }
+
+        /// True if the process has terminated
+    virtual bool isTerminated() = 0;
+        /// Blocks until the process has completed
```

Comment:
Probably not the most important, but we almost want to clarify for stuff like this that it blocks the calling thread (but not the entire process) until the subprocess terminates.